### PR TITLE
feat: add support for redis serverless

### PR DESCRIPTION
**Pull request by syphernl**
## what

- Add support for Serverless Redis instances

## why

- Supporting Redis Serverless for demanding applications

## notes

- This upgrades the required version of the AWS provider from   ">= [5.27.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5270-november-27-2023)" to  ">=  [5.32](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5320-january-11-2024)" as this is the first release where `aws_elasticache_serverless_cache` was introduced

### CODE REVIEW

**Review by Benbentwo — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -96,6 +98,25 @@ locals {
...
   )
+
+  arn = (
+    # If cluster mode is enabled, use the ARN of the replication group
+    local.create_normal_instance ? join("", aws_elasticache_replication_group.default[*].arn) :
+    # If serverless mode is enabled, use the ARN of the serverless cache
+    join("", aws_elasticache_serverless_cache.default[*].arn)
+  )
+
+  endpoint_serverless = try(aws_elasticache_serverless_cache.default[0].endpoint.0.address, null)
```
Comment (Benbentwo) [inline main.tf:109]:
```suggestion
  endpoint_serverless = try(aws_elasticache_serverless_cache.default[0].endpoint[0].address, null)
```

File: main.tf [hunk 2]
```diff
@@ -96,6 +98,25 @@ locals {
...
+  endpoint_cluster    = try(aws_elasticache_replication_group.default[0].configuration_endpoint_address, null)
+  endpoint_instance   = try(aws_elasticache_replication_group.default[0].primary_endpoint_address, null)
+  # Use the serverless endpoint if serverless mode is enabled, otherwise use the cluster endpoint, otherwise use the instance endpoint
+  endpoint_address = coalesce(local.endpoint_serverless, local.endpoint_cluster, local.endpoint_instance)
+
+  reader_endpoint_serverless = try(aws_elasticache_serverless_cache.default[0].reader_endpoint.0.address, null)
```
Comment (Benbentwo) [inline main.tf:115]:
```suggestion
  reader_endpoint_serverless = try(aws_elasticache_serverless_cache.default[0].reader_endpoint[0].address, null)
```

**Review by Benbentwo — CHANGES_REQUESTED** [review]

**Review by Benbentwo — APPROVED** [review]

File: variables.tf [hunk 3]
```diff
@@ -277,3 +277,34 @@ variable "auto_minor_version_upgrade" {
   default     = null
   description = "Specifies whether minor version engine upgrades will be applied automatically to the underlying Cache Cluster instances during the maintenance window. Only supported if the engine version is 6 or higher."
 }
+
+# Add boolean to create a serverless cluster
```
Comment (adamantike) [inline variables.tf:281]: This looks like a leftover comment, used for an AI to generate this variable code :)

**Pull request merged**